extensions/gdata/src/com/google/refine/extension/gdata/UploadCommand.java

```diff
@@ -139,16 +104,70 @@ static private String upload(
             return uploadSpreadsheet(project, engine, params, token, name, exceptions);
         } else if ("gdata/fusion-table".equals(format)) {
             return uploadFusionTable(project, engine, params, token, name, exceptions);
+        } else if (("raw/openrefine-project").equals(format)) {
+            return uploadOpenRefineProject(project, token, name, exceptions);
         }
         return null;
     }
     
+    private static byte[] getImageFromUrl(String urlText) throws IOException {
+        URL url = new URL(urlText);
+        ByteArrayOutputStream output = new ByteArrayOutputStream();
+         
+        try (InputStream inputStream = url.openStream()) {
+            int n = 0;
+            byte [] buffer = new byte[ 1024 ];
+            while (-1 != (n = inputStream.read(buffer))) {
+                output.write(buffer, 0, n);
+            }
+        }
+     
+        return output.toByteArray();
+    }
+
+    private static String uploadOpenRefineProject(Project project, String token,
+            String name, List<Exception> exceptions) {
+        FileOutputStream fos = null;
+        
+        try {
+            java.io.File filePath = java.io.File.createTempFile(name, ".tgz"); 
+            filePath.deleteOnExit();
+            
+            fos = new FileOutputStream(filePath);
+            FileProjectManager.gzipTarToOutputStream(project, fos);
+            
+            File fileMetadata = new File();
+            String asB64 = Base64.encodeBase64URLSafeString(getImageFromUrl(METADATA_ICONLINK));
+            
+            Thumbnail tn = new Thumbnail();
+            tn.setMimeType("image/x-icon").setImage(asB64);
+            ContentHints contentHints = new ContentHints();
+            contentHints.setThumbnail(tn); 
+            
+            fileMetadata.setName(name)
+                .setDescription(METADATA_DESCRIPTION)
+                .setContentHints(contentHints);
+            FileContent projectContent = new FileContent("application/zip", filePath);
+            File file = GoogleAPIExtension.getDriveService(token)
+                    .files().create(fileMetadata, projectContent)
+                .setFields("id")
+                .execute();
+            logger.info("File ID: " + file.getId());
```

Comment:
@jackyq2015 let's mention the method also while logging for easier debugging, like you did before.

Reply:
This is a INFO level logger. should we need that?